### ISSUE TITLE
Foliate just wont start, display is null

### ISSUE THREAD

**Issue by wscharff** [johnfactotum/foliate#1267]
## I just installed Foliate on a fresh Fedora 39 and it wont launch.

1. Go to gnome Software
2. Search for Foliate
3. Install it
4. Execute it -> Does not start

It wont open with flatpak from fedora nor flathub, same for fedora RPM.

Tried to understand what was wrong and running it with `flatpak run com.github.johnfactotum.Foliate` and I got

```
(com.github.johnfactotum.Foliate:2): Gjs-CRITICAL **: 17:54:14.896: JS ERROR: Error: Argument display may not be null
@resource:///com/github/johnfactotum/Foliate/themes.js:110:18


(com.github.johnfactotum.Foliate:2): Gjs-CRITICAL **: 17:54:14.896: Module file:///app/bin/foliate threw an exception
```

Cloned the repo from github and ran `gjs -m src/main.js`. Got the exact same error.

I added `Gtk.init()` after import declarationin main.js and it ran like a charm.

I tried to open the project in Gnome Builder, execute it and it ran like a charm (without the init obviously)

P.S : Yes, I am using dual monitor.

my system is :

## System Details Report
---

### Hardware Information:
- **Hardware Model:**                              MSI MS-7971
- **Memory:**                                      16.0 GiB
- **Processor:**                                   Intel® Core™ i5-6600 × 4
- **Graphics:**                                    NVIDIA GeForce GTX 1060 3GB
- **Disk Capacity:**                               2.7 TB

### Software Information:
- **Firmware Version:**                            1.F0
- **OS Name:**                                     Fedora Linux 39 (Workstation Edition)
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               45.4
- **Windowing System:**                            X11
- **Kernel Version:**                              Linux 6.7.5-200.fc39.x86_64

### Graphics:

![Screenshot from 2024-03-03 18-10-01](https://github.com/johnfactotum/foliate/assets/22889274/bbce77a7-ad4a-4b4b-ab32-08f29617288d)

**Comment by wscharff** [comment]
Here is the difference between Gnome Builder and if I omit that particular line with the gjs interpreter :

![Screenshot from 2024-03-03 22-50-35](https://github.com/johnfactotum/foliate/assets/22889274/f9624bbc-53ff-4e0e-83b8-ecb6e98aeae1)

![image](https://github.com/johnfactotum/foliate/assets/22889274/343e3752-f259-48f6-9ad5-1a586762c39d)

Apparently something is not set and/or ready when the application starts outside Gnome Builder. I reiterate, neither source code, flatpak or RPM works on my system. The call to Gdk.Display.get_default() just returns null in that case. What is strange is that this particular function is called in `app.js:226`, and this time it works.

=> Calling Gtk.init() in main.js genuinely seems to fix the issue

**Comment by johnfactotum** [comment]
The more "correct" fix would be adding the style provider after GTK is initialized, i.e. in the handler for the `startup` signal. Not sure why it works now, though.

**Comment by SylChamber** [comment]
I have the exact same error (NVidia graphics) since my KDE Neon upgraded itself to KDE Plasma 6 with Wayland session by default. I will wait for the fix.

**Comment by SylChamber** [comment]
I confirm that it was fixed in the latest Flatpak version, 3.1.1, which was updated yesterday.
